File: Seminar7/DZ/java/service/UserServiceImpl.java
package Seminar7.DZ.java.service;

import java.util.List;

import Seminar7.DZ.java.model.User;
import Seminar7.DZ.java.repository.UserRepository;

public class UserServiceImpl implements UserService {

	private UserRepository userRepository = new UserRepository();

	@Override
	public void saveUser(User user) {
		if (user != null) {
			userRepository.saveUser(user);
		} else {
			System.out.println("User is null");
		}
	}

	@Override
	public void deleteUser(User user) {
		if (user != null) {
			userRepository.deleteUser(user);
		} else {
			System.out.println("User is null");
		}
	}

	@Override
	public void updateUser(int index, User user) {
		if (user != null) {
			userRepository.updateUser(index, user);
		} else {
			System.out.println("User is null");
		}
	}

	@Override
	public List<User> getAllUsers() {
		return userRepository.getAllUsers();
	}

	@Override
	public void addFunds(int index, float funds) {
		userRepository.addFunds(index, funds);
	}

	@Override
	public void removeFunds(int index, float funds) {
		userRepository.removeFunds(index, funds);
	}

}

File: Seminar7/DZ/py/UserServiceImpl.py
from UserService import UserService
from UserRepository import UserRepository


class UserServiceImpl(UserService):
    __userRepository = UserRepository()

    def __init__(self):
        self.__userRepository = UserRepository()

    def saveUser(self, user):
        if user is not None:
            self.__userRepository.saveUser(user)
        else:
            print("User is null")

    def deleteUser(self, user):
        if user is not None:
            self.__userRepository.deleteUser(user)
        else:
            print("User is null")

    def updateUser(self, index, user):
        if user is not None:
            self.__userRepository.updateUser(index, user)
        else:
            print("User is null")

    def getAllUsers(self):
        return self.__userRepository.getAllUsers()

    def addFunds(self, index, funds):
        self.__userRepository.addFunds(index, funds)

    def removeFunds(self, index, funds):
        self.__userRepository.removeFunds(index, funds)
